runtime/lua/vim/img.lua

```diff
@@ -0,0 +1,78 @@
+local img = vim._defer_require('vim.img', {
+  _backend = ..., --- @module 'vim.img._backend'
+  _detect = ..., --- @module 'vim.img._detect'
+  _image = ..., --- @module 'vim.img._image'
+  _terminal = ..., --- @module 'vim.img._terminal'
+})
+
+---Loads an image into memory, returning a wrapper around the image.
+---
+---Accepts `data` as base64-encoded bytes, or a `filename` that will be loaded.
+---@param opts {data?:string, filename?:string}
+---@return vim.img.Image
+function img.load(opts)
+  return img._image:new(opts)
+end
+
+img.protocol = (function()
+  ---@class vim.img.Protocol 'iterm2'|'kitty'|'sixel'
+
+  ---@type vim.img.Protocol|nil
+  local protocol = nil
+
+  local loaded = false
+
+  ---Determines the preferred graphics protocol to use by default.
+  ---
+  ---@return vim.img.Protocol|nil
+  return function()
+    if not loaded then
+      local graphics = img._detect().graphics
+
+      ---@diagnostic disable-next-line:cast-type-mismatch
+      ---@cast graphics vim.img.Protocol|nil
+      protocol = graphics
+
+      loaded = true
+    end
+
+    return protocol
+  end
+end)()
+
+---@class vim.img.Opts: vim.img.Backend.RenderOpts
+---@field backend? vim.img.Protocol|vim.img.Backend
+
+---Displays the image within the terminal used by neovim.
```

Comment:
Let's try to orient around "UI", not "terminal", where possible. This might only support terminals for now. But there is really no need to mention "neovim" etc here anyway.

```suggestion
--- Displays an image.
```